otakaro/processes/geometry.py

```diff
@@ -0,0 +1,99 @@
+# # -*- coding: utf-8 -*-
+"""Defines PyWPS WebProcessingService process for marking points and calculating polygon areas."""
+
+import json
+
+import geopandas as gpd
+from pywps import Process, Format, ComplexInput, LiteralOutput, WPSRequest
+from pywps.response.execute import ExecuteResponse
+
+
+class GeometryProcessService(Process):
+    """Class representing a WebProcessingService process for marking points and calculating polygon areas."""
+
+    def __init__(self) -> None:
+        """Define inputs and outputs of the WPS process, and assign process handler."""
+        # Create point or polygon WPS inputs
+        inputs = [
+            ComplexInput('location', 'Location',
+                         supported_formats=[
+                             Format(mime_type='application/vnd.geo+json',
+                                    schema='http://geojson.org/geojson-spec.html#geojson')],
+                         workdir='workdir'
+                         )
+        ]
+        # Create lat and lon or area WPS outputs
+        outputs = [
+            LiteralOutput('details', 'Details', data_type='string')
+        ]
+
+        # Initialise the process
+        super().__init__(
+            self._handler,
+            identifier='geometry',
+            title="Mark Points and Calculate Polygon Areas",
+            inputs=inputs,
+            outputs=outputs
+        )
+
+    @staticmethod
+    def _handler(request: WPSRequest, response: ExecuteResponse) -> None:
+        """
+        Process handler for marking points and calculating polygon areas.
+
+        Parameters
+        ----------
+        request : WPSRequest
+            The WPS request, containing input parameters.
+        response : ExecuteResponse
+            The WPS response, containing output data.
+        """
+        # Get the location input (GeoJSON string) from the request
+        layer_input = request.inputs["location"][0].data
+        # Parse the GeoJSON string into a dictionary
+        layer_json = json.loads(layer_input)
+
+        try:
+            # Get the geometry type of the feature
+            geom_type = layer_json["features"][0]["geometry"]["type"]
+        except IndexError:
+            # Set the geometry type to 'Unknown' if it is not found or an error occurs
+            geom_type = "Unknown"
```

Comment:
No need to make changes now you could probably read this with geopandas from the beginning